File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //the key is the missing sum, the elem is the index
        HashMap<Integer,Integer> myMap = new HashMap<>();
        int len = nums.length;
        int[] returnArr = new int[2];
        for(int i = 0; i < len ; i++){
            if(myMap.containsKey(target - nums[i])){
                returnArr[0] = i;
                returnArr[1] = myMap.get(target - nums[i]); //the elem is the index
                return returnArr;
            }
            myMap.put(nums[i],i);
        }
        return returnArr;
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        "make hashtable where the the key is target-nums[i] and value is the index "
        "if target-nums[i] dosent exist , insert nums[i] to hashtable"
        my_dict = {}
        solution = [0,0]
        for i in range(len(nums)):
            if target-nums[i] in my_dict:
                solution[0] = i
                solution[1] = my_dict[(target-nums[i])]
                return solution
            else:
                my_dict[nums[i]] = i
        return solution
